internal/pkg/template/templates/task/cf.yml

```diff
@@ -96,7 +96,7 @@ Resources:
             Action: 'sts:AssumeRole'
       ManagedPolicyArns:
         - !Sub 'arn:${AWS::Partition}:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
-      {{- if .Secrets}}
```

Comment:
Just to double check, Is the template still correct with the combination of condition
```
- !If
  - HasAppAndEnv
  - Policy
  - !Ref AWS::NoValue
```
and the new nested ``{{ If }}``s ?

Reply:
So just to explain -- 
if we have App and Env then we will add a Policy else nothing will be added.
On the other hand if App and Env is not provided then we will prompt permission question and that flow will build SSMParamSecrets and secretsManagerSecrets maps and if those are present then nested if logics will work and we will add specific policy statements

Reply:
We are using both "Condition" and go-template rendering (i.e. `{{ If }}` ) to implement a `if-else` logic. Can we simplify it so that we use either "Condition" or go-template rendering? This may seem confusing for future readers.

Reply:
For example
```yaml
{{-If .Secrets}}
  {{- if and .App .Env}}
  {{- else}}
  {{- end}}
{{- end}}
```